ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -335,10 +336,28 @@ private void checkCapacityAndUpdateWriteStatusDelegate(long totalCapacity, long
       }
       //else: maintain current replicaId status if percentFilled between threshold - delta and threshold
     } else {
-      logger.debug("The WriteStatusDelegate is not instantiated");
+      logger.debug("The ReplicaStatusDelegate is not instantiated");
     }
   }
 
+  /**
+   * Update the stopped replicas list based on the state of this store.
+   * @param isStopped whether the store is stopped ({@code true}) or started.
+   * @return {@code true} if StoppedReplicas list has been updated successfully.
+   */
+  boolean updateStoppedReplicasList(boolean isStopped) {
```

Comment:
if we are still stopping one store at a time, wouldn't it defeat the purpose of having a batch API? How do we leverage the batch API?

cc: @pnarayanan

Reply:
Since this method is in `BlobStore.java`, I think it naturally serves a single blobstore. If we need a batch process, we can directly invoke `ReplicaStatusDelegate.markStopped(List<ReplicaId> replicaIds)` because both `StorageManager` and `DiskManager` can access methods in`ReplicaStatusDelegate`.

Reply:
The last time @jsjtzyy and I discussed this, a potential use case was marking all stores on a disk as stopped, when there is a disk failure. Like @jsjtzyy says, this will require the `DiskManager` or `StorageManager` invoking the api directly.